sci-mathematics/msieve/msieve-1.53.recipe

```diff
@@ -0,0 +1,66 @@
+SUMMARY="Library for Factoring Large Integers"
+DESCRIPTION="C library implementing a suite of algorithms to factor large integers"
+HOMEPAGE="https://sourceforge.net/projects/msieve/"
+COPYRIGHT="Public Domain"
+LICENSE="Public Domain"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/msieve/msieve/Msieve%20v$portVersion/msieve${portVersion/./}_src.tar.gz"
+CHECKSUM_SHA256="c5fcbaaff266a43aa8bca55239d5b087d3e3f138d1a95d75b776c04ce4d93bb4"
+PATCHES="msieve-1.53.patchset"
+
+ARCHITECTURES="?x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	msieve$secondaryArchSuffix = $portVersion
+	cmd:msieve$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libecm$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	msieve${secondaryArchSuffix}_devel = $portVersion
+	devel:libmsieve$secondaryArchSuffix
+	"
+REQUIRES_devel="
+	haiku${secondaryArchSuffix}_devel
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libecm$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:sed
+	"
+
+PATCH()
+{
+	# Allow owns CFLAGS, could be useful when doing debug builds
+	sed -e 's/^CFLAGS =/CFLAGS +=/' -i Makefile
+}
+
+BUILD()
+{
+	make $jobArgs all ECM=1 HAIKU=1
+}
+
+INSTALL()
+{
+	install -d -m 755 "$binDir" "$docDir" "$includeDir" "$developLibDir"
+
+	install -t "$binDir" -m 755 msieve
+	install -t "$docDir" -m 644 Changes Readme*
+	install -t "$includeDir" -m 644 include/*
+	install -t "$developLibDir" -m 644 libmsieve.a
+
+	packageEntries devel "$developDir"
```

Comment:
Hmm, the files Changes would be better in the devel package, after all. What do you think?
```diff
-	packageEntries devel "$developDir"
+	packageEntries devel \
+		"$developDir" \
+		"$docDir"/Changes
```

Reply:
`msieve` is more or less a demo for `libmsieve` anyway (the source file is literally `demo.c`). So I thing pretty much every file in `$docDir` should end up in the devel package. What's your opinion on this?

Reply:
Although I do agree that the information in the Readme files will be useful mostly to developers, having these files in the base package is slightly better because it shows that we provide credits to the authors.
Indeed, moving the Readme files to the devel package might be seen as an attempt to hide the credits.
If I were one of the authors I would wish to have the Readme files in the base package.
Moreover, these files are not very big ;-)
(For the Changes file, that's another story, since this can be seen as meta-data.)